pkg/execution/state/redis_state/queue.go

```diff
@@ -564,42 +565,82 @@ func (q QueueShard) Partition() string {
 	return q.Name
 }
 
+// FnMetadata is stored within the queue for retrieving
+type FnMetadata struct {
+	// NOTE: This is not encoded via JSON as we should always have the function
+	// ID prior to doing a lookup, or should be able to retrieve the function ID
+	// via the key.
+	// FnID uuid.UUID `json:"-"` // TODO: Do we need this?
+
+	// Paused represents whether the fn is paused.  This allows us to prevent leases
+	// to a given partition if the partition belongs to a fn.
+	Paused bool `json:"off"`
+}
+
 // QueuePartition represents an individual queue for a workflow.  It stores the
 // time of the earliest job within the workflow.
 type QueuePartition struct {
+	// TODO: CAN WE REMOVE THIS?
 	QueueName *string `json:"queue,omitempty"`
 
-	WorkflowID  uuid.UUID `json:"wid"`
-	WorkspaceID uuid.UUID `json:"wsID"`
-
+	// PartitionType is the int-value of the enums.PartitionType for this
+	// partition.  By default, partitions are function-scoped without any
+	// custom keys.
+	PartitionType int `json:"pt,omitempty"`
+	// ConcurrencyScope is the int-value representation of the enums.ConcurrencyScope,
+	// if this is a concurrency-scoped partition.
+	ConcurrencyScope int `json:"cs,omitempty"`
+	// FunctionID represents the function ID that this partition manages.
+	// NOTE: This may be nil for account and environment-scoped concurrency
+	// keys which make partitions of many functions.
+	FunctionID *uuid.UUID `json:"wid,omitempty"`
+	// EnvID represents the environment ID for the partition, either from the
+	// function ID or the environment scope itself.
+	EnvID *uuid.UUID `json:"wsID,omitempty"`
+	// AccountID represents the account ID for the partition.  This ONLY exists
+	// if the partition represents an account-level concurrency key.
+	AccountID *uuid.UUID `json:"aID,omitempty"`
+	// Priority represents the partition's priority.  This currently exists
+	// on the partition (instead of fn metadata) to reduce lookups when peeking
+	// many partitions.
 	Priority uint `json:"p"`
-	Paused   bool `json:"off"`
```

Comment:
nb. requires app migration for deploy

Reply:
https://linear.app/inngest/issue/INN-3236/migrations-for-partition-paused-→-fn-paused

Reply:
Done.  Thanks!